src/modals/storage-connections/StorageConnectionsTable.js

```diff
@@ -0,0 +1,356 @@
+    const rows = []
+    connections.forEach((connection, index) => {
+      rows.push(createRow(type, connection, true, index + 1))
+    })
+    if (isNewConnectionOpened) {
+      rows.unshift(createRow(type, null, false, 0))
+      rows[0] = applyCellEdits(rows[0], 'edit', false)
+    }
+    return rows
+  }
+
+  const createRow = (type, connection, areActionsEnabled, rowIndex) => {
+    const createCells = cellsCreator[type]
+    return {
+      cells: createCells(connection, rowIndex),
+      connection: connection,
+      isHoverable: true,
+      disableActions: !areActionsEnabled,
+      rowEditValidationRules: validationRules,
+    }
+  }
+
+  const actionResolver = (rowData, _) => {
+    return [
+      {
+        title: msg.storageConnectionsAttachConnectionButton(),
+        onClick: (event, rowId, rowData, extra) => attachConnection(rowData.connection),
+        isDisabled: rowData.connection && !canAttachConnection(rowData.connection, storageDomain),
+      },
+      {
+        title: msg.storageConnectionsDetachConnectionButton(),
+        onClick: (event, rowId, rowData, extra) => detachConnection(rowData.connection),
+        isDisabled: rowData.connection && !canDetachConnection(rowData.connection, storageDomain),
+      },
+      {
+        title: msg.storageConnectionsRemoveConnectionButton(),
+        onClick: (event, rowId, rowData, extra) => deleteConnection(rowData.connection),
+        isDisabled: rowData.connection && !canRemoveConnection(rowData.connection),
+      },
+    ]
+  }
+
+  const requiredRule = {
+    name: 'required',
+    validator: val => val.trim() !== '',
+    errorText: msg.storageConnectionsFieldRequiredError(),
+  }
+
+  const portRule = {
+    name: 'port',
+    validator: val => val > 0 && val <= 0xFFFF,
+    errorText: msg.storageConnectionsFieldPortError(),
+  }
+
+  const validationRules = [requiredRule, portRule]
+
+  const [rows, setRows] = useState(createRows(type, connections))
+  useEffect(() => {
+    const newRows = createRows(type, connections)
+    setRows(newRows)
+  }, [connections, isNewConnectionOpened])
+
+  return (
+    <Table
+      aria-label='Storage Connections Table'
+      className='storage-connections-table'
+      cells={getColumns()}
+      rows={rows}
+      onRowEdit={updateRows}
+      variant='compact'
+      actionResolver={actionResolver}
+      areActionsDisabled={rowData => !!rowData.disableActions}
+      dropdownPosition="left"
+      dropdownDirection="down"
+    >
+      <TableHeader />
+      <TableBody />
+    </Table>
+  )
+}
+
+const canEditConnection = (connection, storageDomain) => {
+  if (connection) {
+    return !connection.isAttachedToDomain || storageDomain.status === 'maintenance'
+  } else {
+    return false
+  }
+}
+
+const canAttachConnection = (connection, storageDomain) => {
+  if (connection) {
+    return !connection.isAttachedToDomain && storageDomain.status === 'maintenance'
+  } else {
+    return false
+  }
+}
```

Comment:
is it possible to do:
```suggestion
    return connection && !connection.isAttachedToDomain && storageDomain.status === 'maintenance'
```

Reply:
Same for "canEditConnection", "canDetachConnection" & "canRemoveConnection".
Another option is to fail first and then no need for "else":
if (!connection) {
  return false
}